test/unit/mdc-list/foundation.test.js

```diff
@@ -652,3 +691,27 @@ test('#setUseActivatedClass causes setSelectedIndex to use the --activated class
 
   td.verify(mockAdapter.addClassForElementIndex(1, cssClasses.LIST_ITEM_ACTIVATED_CLASS), {times: 1});
 });
+
+test('#setSelectedIndex should bail out if not in the range', () => {
+  const {foundation, mockAdapter} = setupTest();
+
+  td.when(mockAdapter.getListItemCount()).thenReturn(4);
+  foundation.setSelectedIndex(-1);
+  td.verify(mockAdapter.setAttributeForElementIndex(-1, 'tabindex', 0), {times: 0});
+});
+
+test('#setSelectedIndex should set aria attributes on new index and should not change aria attributes on previous' +
```

Comment:
This is specifically for checkbox, right? Can we include checkbox somewhere in the test description? And do we have a similar test for radio / single select?